docs/_integration-with-existing-apps-kotlin.md

```diff
@@ -119,257 +106,359 @@ repositories {
-}
-```
+class MyReactActivity : ReactActivity() {
 
-We also need to pass back button events to React Native:
+    override fun getMainComponentName(): String = "HelloWorld"
 
-```kotlin
-override fun onBackPressed() {
-    reactInstanceManager.onBackPressed()
-    super.onBackPressed()
+    override fun createReactActivityDelegate(): ReactActivityDelegate =
+        DefaultReactActivityDelegate(this, mainComponentName, fabricEnabled)
 }
 ```
 
-This allows JavaScript to control what happens when the user presses the hardware back button (e.g. to implement navigation). When JavaScript doesn't handle the back button press, your `invokeDefaultOnBackPressed` method will be called. By default this finishes your `Activity`.
+</TabItem>
+</Tabs>
 
-Finally, we need to hook up the dev menu. By default, this is activated by (rage) shaking the device, but this is not very useful in emulators. So we make it show when you press the hardware menu button (use <kbd>Ctrl</kbd> + <kbd>M</kbd> if you're using Android Studio emulator):
+As usual, here the [MainActivity.kt Community template file as reference](https://github.com/react-native-community/template/blob/main/template/android/app/src/main/java/com/helloworld/MainApplication.kt)
 
-```kotlin
-override fun onKeyUp(keyCode: Int, event: KeyEvent?): Boolean {
-    if (keyCode == KeyEvent.KEYCODE_MENU && reactInstanceManager != null) {
-        reactInstanceManager.showDevOptionsDialog()
-        return true
-    }
-    return super.onKeyUp(keyCode, event)
-}
+Whenever you create a new Activity, you need to add it to your `AndroidManifest.xml` file. You also need set the theme of `MyReactActivity` to `Theme.AppCompat.Light.NoActionBar` (or to any non-ActionBar theme) as otherwise your application will render an ActionBar on top of your React Native screen:
+
+```diff
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <uses-permission android:name="android.permission.INTERNET" />
+
+    <application
+      android:name=".MainApplication">
+
++     <activity
++       android:name=".MyReactActivity"
++       android:label="@string/app_name"
++       android:theme="@style/Theme.AppCompat.Light.NoActionBar">
++     </activity>
+    </application>
+</manifest>
 ```
 
 Now your activity is ready to run some JavaScript code.
 
-### Test your integration
+## 6. Test your integration
+
+You have now done all the basic steps to integrate React Native with your current application. Now we will start the [Metro bundler](https://metrobundler.dev/) to build the TypeScript code of your application and have the server running on localhost to serve it.
 
-You have now done all the basic steps to integrate React Native with your current application. Now we will start the [Metro bundler][metro] to build the `index.bundle` package and the server running on localhost to serve it.
+First, you need to create a `metro.config.js` file in the root of your project as follows:
```

Comment:
from my tests, that's not enough.
You also have to install the `@react-native/metro-config` package and you have to create a `.watchmanconfig` fle with `{}` as content. Otherwise the user will incur in errors.

Reply:
I've just tested in the Android app and I don't need it at all (not sure if this is a requirement once you're not on a Meta laptop).